src/Tribe/I18n.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Handles internationalization of The Events Calendar strings.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events
+ */
+
+namespace Tribe\Events;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Events__Main as TEC;
+use Tribe__Main as Common;
+
+/**
+ * Class I18n
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events
+ */
+class I18n {
+	/**
+	 * An instance of the The Events Calendar main class.
+	 *
+	 * @since TBD
+	 *
+	 * @var TEC
+	 */
+	protected $tec;
+
+	/**
+	 * I18n constructor.
+	 *
+	 * @param TEC $tec An instance of The Events Calendar main class.
+	 */
+	public function __construct( TEC $tec ) {
+		$this->tec = $tec;
+	}
+
+	/**
+	 * Get all possible translations for a String based on the given Languages and Domains
+	 *
+	 * WARNING: This function is slow because it deals with files, so don't overuse it!
+	 *
+	 * @since TBD Moved here from Tribe__Events__Main.
+	 *
+	 * @param array  $strings          An array of strings (required).
+	 * @param array  $languages        Which l10n to fetch the string (required).
+	 * @param array  $domains          Possible Domains to re-load.
+	 * @param string $default_language The default language to avoid re-doing that.
+	 *
+	 * @return array                    A multi level array with the possible translations for the given strings
+	 */
+	public function get_i18n_strings( $strings, $languages, $domains = array(), $default_language = 'en_US' ) {
+		$domains = wp_parse_args( $domains, array(
+			'default'             => true, // Default doesn't need file path
+			'the-events-calendar' => $this->tec->plugin_dir . 'lang/',
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 9c82d7b7e3912ec89f0373f3df752017</sub>